src/components/DataSubmissions/DeleteAllOrphanFilesButton.test.tsx

```diff
@@ -0,0 +1,358 @@
+
+  beforeEach(() => {
+    onDelete.mockReset();
+  });
+
+  it("should render default chip with label and icon", () => {
+    const { getByTestId } = render(
+      <TestParent context={{ ...baseContext, user: { ...baseUser, role: "Admin" } }}>
+        <DeleteAllOrphanFilesButton submission={baseSubmission} onDelete={onDelete} />
+      </TestParent>
+    );
+
+    expect(getByTestId("delete-all-orphan-files-icon")).toBeDefined();
+    expect(getByTestId("delete-all-orphan-files-button")).not.toBeDisabled();
+  });
+
+  it("should open delete dialog when the button when is clicked", async () => {
+    const { getByTestId } = render(
+      <TestParent
+        context={{ ...baseContext, user: { ...baseUser, role: "Admin" } }}
+        mocks={successMocks}
+      >
+        <DeleteAllOrphanFilesButton submission={baseSubmission} onDelete={onDelete} />
+      </TestParent>
+    );
+
+    userEvent.click(getByTestId("delete-all-orphan-files-button"));
+
+    await waitFor(() => {
+      expect(screen.getByText("Delete All Orphaned Files")).toBeInTheDocument();
+    });
+  });
+
+  it("should disable the button when in loading state", async () => {
+    const { getByTestId } = render(
+      <TestParent
+        context={{ ...baseContext, user: { ...baseUser, role: "Admin" } }}
+        mocks={successMocks}
+      >
+        <DeleteAllOrphanFilesButton submission={baseSubmission} onDelete={onDelete} />
+      </TestParent>
+    );
+
+    userEvent.click(getByTestId("delete-all-orphan-files-button"));
+
+    await waitFor(() => {
+      expect(screen.getByText("Delete All Orphaned Files")).toBeInTheDocument();
+    });
+
+    userEvent.click(getByTestId("delete-dialog-confirm-button"));
+
+    expect(getByTestId("delete-all-orphan-files-button")).toBeDisabled();
+  });
+
+  it("should disable the button when disabled prop is passed", () => {
+    const { getByTestId } = render(
+      <TestParent
+        context={{ ...baseContext, user: { ...baseUser, role: "Admin" } }}
+        mocks={successMocks}
+      >
+        <DeleteAllOrphanFilesButton submission={baseSubmission} onDelete={onDelete} disabled />
+      </TestParent>
+    );
+
+    expect(getByTestId("delete-all-orphan-files-button")).toBeDisabled();
+  });
+
+  it("should call onDelete with true on successful mutation", async () => {
+    const { getByTestId } = render(
+      <TestParent
+        context={{ ...baseContext, user: { ...baseUser, role: "Admin" } }}
+        mocks={successMocks}
+      >
+        <DeleteAllOrphanFilesButton submission={baseSubmission} onDelete={onDelete} />
+      </TestParent>
+    );
+
+    userEvent.click(getByTestId("delete-all-orphan-files-button"));
+
+    await waitFor(() => {
+      expect(screen.getByText("Delete All Orphaned Files")).toBeInTheDocument();
+    });
+
+    userEvent.click(getByTestId("delete-dialog-confirm-button"));
+
+    await waitFor(() => {
+      expect(onDelete).toHaveBeenCalledWith(true);
+    });
+  });
```

Comment:
I think this case is testing the same thing as below? The only difference I see is the missing alert assertion in this one. No action necessarily needed here, just wanted to point it out in case it's an accident.

Reply:
Removed test since it is already covered by another test. Thanks!